mpp/include/mpp_do_update_ad.fh

```diff
@@ -21,8 +19,7 @@
 !***********************************************************************
 !> @addtogroup mpp_domains_mod
 !> @{
-
-    !> Updates data domain of 3D field whose computational domains have been computed
+    !> Updates data domain of 3D ad field whose computational domains have been computed
```

Comment:
Please add a more detailed description of what this routine is doing.  I understand that the description that was here was very small, so it would be great to have it updated with more detail.  Comments such as "what a forward array is" would be very helpful for developers learning this routine.

@jkbk2004 you are the most knowledgeable person with insight into what this routine is doing.  The entire FMS team would appreciate a longer detailed description.  Most of the new code we are writing includes better descriptions of our routines, so this request is in coordination with our larger effort to improve documentation.

Reply:
Added more description and hopefully Beny's short paper might be helpful for future developers.